frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -104,6 +107,28 @@ export const AddCapacityModal = (props: AddCapacityModalProps) => {
     }
   };
 
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    setProgress(true);
+
+    if (isNoProvionerSC) {
+      const [pvsData, pvsLoaded, pvsLoadError] = pvsList;
+      if ((pvsLoadError || pvsData.length === 0 || !storageClass) && pvsLoaded) {
+        // if pv list fails, pass the size as 1
+        makeAddCapacityRequest(defaultRequestSize.BAREMETAL);
+      } else if (pvsLoaded) {
+        const pvs = getSCAvailablePVs(pvsData, getName(storageClass));
+        // empty pvs list will return infinity, hence added this condition
+        const size = pvs.length
+          ? getMinSizePVCapacity(pvs).toString()
+          : defaultRequestSize.BAREMETAL;
```

Comment:
PvsLoaded and pvs.length ===0 will be caught in first block itself. Why we need to check again?

Reply:
This is applicable when the pvs with available state are not there. There can be other pvs with bound state and hence will come to this block.